### PR TITLE
Added packaging script for webextension

### PR DESCRIPTION
**Pull request by cohosh**
Added a new script to package the webextension. This will automatically
build and zip the source code and the webextension for upload. It take a
version as an argument and then checks the version in the manifest, and
locally commits a version bump.

### CODE REVIEW

File: proxy/make.js [hunk 1]
```diff
@@ -112,6 +112,30 @@ task('node', 'build the node binary', function() {
...
+    execSync(`rm -f webext/webext.zip`);
+  } catch (error) {
+    console.log('Error removing zip files');
+  }
+  execSync(`git submodule update --remote`);
+  var version = process.argv[3];
+  console.log(version);
+  var manifest = require('./webext/manifest.json')
+  manifest.version = version;
+  writeFileSync('./webext/manifest.json', JSON.stringify(manifest, null, 2), 'utf8');
```
Comment (NullHypothesis) [inline proxy/make.js]: I think I'm missing something here but manifest.json contains the line `"description": "__MSG_appDesc__",`. Isn't this supposed to be substituted by a (translated) string? When I run `node make.js pack-webext foo`, the resulting webext.zip contains the original manifest.json, including `__MSG_appDesc__`.

File: proxy/make.js [hunk 2]
```diff
@@ -112,6 +112,30 @@ task('node', 'build the node binary', function() {
...
+  execSync(`git commit -am "bump version to ${version}"`);
+  try {
+    execSync(`git tag webext-${version}`);
+  } catch (error) {
+    console.log('Error creating git tag');
```
Comment (NullHypothesis) [inline proxy/make.js]: Do we really want try/catch blocks? That is, would we want to upload a new version even though it doesn't have a git tag?

File: proxy/make.js (hunk 1 above)
Comment (cohosh) [reply proxy/make.js]: The description is a localizable property of the manifest. See https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Internationalization#Internationalizing_manifest.json

File: proxy/make.js (hunk 2 above)
Comment (cohosh) [reply proxy/make.js]: This is partially why I don't have the script automatically push. I figured the script should display the error, and then the user can either fix the error manually or revert the git commit and re-run the script.

I could also have the catch block revert the git commit and return, but either way I think catching the error and responding to it is strictly better than just letting node fail.

Comment (cohosh) [reply proxy/make.js]: Alright, I added a new commit to undo all the changes this script does in the case that a `git tag` fails.

**Comment by cohosh** [comment]
This was merged.

**Pull request closed without merging**